tremor-script/src/lexer.rs

```diff
@@ -1824,300 +1824,270 @@ impl<'input> Lexer<'input> {
+                )
+            };
+            res.push(self.spanned2(*segment_start, end_inner, token));
+            *has_escapes = false;
+        }
+        *segment_start = end_inner;
+        res.push(self.spanned2(*segment_start, *end, Token::Interpol));
+        let mut pcount = 0;
         loop {
-            let ch = self.bump();
-            match ch {
-                Some((e, '\n')) => {
-                    end = e;
-                    heredoc_content.push('\n');
-                    res.push(self.spanned2(
-                        segment_start,
-                        end,
-                        Token::StringLiteral(heredoc_content.into()),
-                    ));
-                    end.shift('\n');
-                    segment_start = end;
-                    heredoc_content = String::new();
-                }
-                Some((end_inner, '\\')) => {
-                    let (mut e, c) = self.escape_code(&heredoc_start, end_inner)?;
-                    if c == '{' || c == '}' {
-                        if !heredoc_content.is_empty() {
-                            let token = if has_escapes {
-                                // The string was modified so we can't use the slice
-                                Token::StringLiteral(heredoc_content.into())
-                            } else {
-                                self.slice(segment_start, end_inner).map_or_else(
-                                    || Token::StringLiteral(heredoc_content.into()),
-                                    |slice| Token::StringLiteral(slice.into()),
-                                )
-                            };
-                            res.push(self.spanned2(segment_start, end_inner, token));
-                            heredoc_content = String::new();
-                            has_escapes = false;
+            match self.next() {
+                Some(Ok(s)) => {
+                    match &s.value {
+                        Token::RBrace if pcount == 0 => {
+                            *segment_start = s.span.pp_end;
+                            res.push(s);
+                            break;
                         }
-                        if c == '{' {
-                            res.push(self.spanned2(end_inner, e, Token::QuotedSquigglyOpen));
-                        } else {
-                            res.push(self.spanned2(end_inner, e, Token::QuotedSquigglyClose));
-                        };
-                        e.shift(c);
-                        segment_start = e;
-                        end = e;
-                    } else {
-                        has_escapes = true;
-                        heredoc_content.push(c);
-                        end = e;
-                        end.shift(c);
-                    }
+                        Token::RBrace => {
+                            pcount -= 1;
+                        }
+                        Token::LBrace | Token::Interpol => {
```

Comment:
do we really need `Token::LBrace` here anymore? `Interpol` should suffice, no?

Reply:
We do since we need to count them in something like:

```tremor
"""
this is a #{  { "here": "doc" } }!
"""
```